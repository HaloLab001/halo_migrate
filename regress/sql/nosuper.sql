--
-- no superuser check
--
SET client_min_messages = error;
DROP ROLE IF EXISTS nosuper;
SET client_min_messages = warning;
CREATE ROLE nosuper WITH LOGIN;
-- => OK
\! pg_migrate --execute --alter='ADD COLUMN ns1 INT' --dbname=contrib_regression --table=tbl_cluster --no-superuser-check
-- => ERROR
\! pg_migrate --execute --alter='ADD COLUMN ns2 INT' --dbname=contrib_regression --table=tbl_cluster --username=nosuper
-- => ERROR
\! pg_migrate --execute --alter='ADD COLUMN ns3 INT' --dbname=contrib_regression --table=tbl_cluster --username=nosuper --no-superuser-check
DROP ROLE IF EXISTS nosuper;
