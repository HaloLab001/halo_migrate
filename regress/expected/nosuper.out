--
-- no superuser check
--
SET client_min_messages = error;
DROP ROLE IF EXISTS nosuper;
SET client_min_messages = warning;
CREATE ROLE nosuper WITH LOGIN;
-- => OK
\! halo_migrate --execute --alter='ADD COLUMN ns1 INT' --dbname=contrib_regression --table=tbl_cluster --no-superuser-check
INFO: migrating table "public.tbl_cluster"
INFO: altering table with: ADD COLUMN ns1 INT
-- => ERROR
\! halo_migrate --execute --alter='ADD COLUMN ns2 INT' --dbname=contrib_regression --table=tbl_cluster --username=nosuper
ERROR: halo_migrate failed with error: You must be a superuser to use halo_migrate
-- => ERROR
\! halo_migrate --execute --alter='ADD COLUMN ns3 INT' --dbname=contrib_regression --table=tbl_cluster --username=nosuper --no-superuser-check
ERROR: halo_migrate failed with error: ERROR:  permission denied for schema migrate
LINE 1: select migrate.version(), migrate.version_sql()
               ^
DROP ROLE IF EXISTS nosuper;
